RELEASE_CHECKLIST.md

```diff
@@ -0,0 +1,111 @@
+# Release Checklist
+
+This is a list of the things that need to happen during a release.
+
+## Build a Release
+
+### Prepare the Changelog (Full release only)
+
+1. Open the associated milestone. All issues and PRs should be closed. If
+   they are not you should reassign all open issues and PRs to future
+   milestones.
+1. Go through the commit history since the last release. Ensure that all PRs
+   that have landed are marked with the milestone. You can use this to
+   show all the PRs that are merged on or after YYY-MM-DD:
+   `https://github.com/issues?q=repo%3Acloudflare%2Fkv-asset-handler+merged%3A%3E%3DYYYY-MM-DD`
+1. Go through the closed PRs in the milestone. Each should have a changelog
+   label indicating if the change is docs, fix, feature, or maintenance. If
+   there is a missing label, please add one.
+1. Choose an emoji for the release. Try to make it semi-related to something that's been included in the release (point releases can be a little weirder).
+1. Add this release to the `CHANGELOG.md`. Use the structure of previous
+   entries. If you use VS Code, you can use [this snippet](https://gist.github.com/EverlastingBugstopper/04d1adb99506388ff9d7abd8d0a82bc3) to insert new changelog sections. If it is a release candidate, no official changelog is needed, but testing instructions will be added later in the process.
+
+### Start a release PR
+
+1. Create a new branch "#.#.#" where "#.#.#" is this release's version (release) or "#.#.#-rc.#" (release candidate)
+1. Push up a commit with the `CHANGELOG.md` changes. The commit message can just be "#.#.#" (release) or "#.#.#-rc.#" (release candidate)
+1. Request review from the @cloudflare/workers-devexp team.
+
+### Review
+
+Most of your comments will be about the changelog. Once the PR is finalized and approved...
+
+1. If you made changes, squash or fixup all changes into a single commit.
+1. Run `git push` and wait for CI to pass.
+
+### Tag and build release
+
+This part of the release process is handled by GitHub Actions, and our binaries are distributed as GitHub Releases. When you push a version tag, it kicks off an action that creates a new GitHub release for that tag, builds release binaries and attaches them to the release.
```

Comment:
is this true for this repo? we don't build binaries, we just publish the package i think, so maybe we should remove copy that references binaries.